src/Nethermind/Nethermind.Consensus/Tracing/GethStyleTracer.cs

```diff
@@ -172,7 +172,7 @@ public IEnumerable<string> TraceBlockToFile(Hash256 blockHash, GethTraceOptions
     private IBlockTracer<GethLikeTxTrace> CreateOptionsTracer(BlockHeader block, GethTraceOptions options) =>
         options switch
         {
-            { Tracer: var t } when GethLikeNativeTracerFactory.IsNativeTracer(t) => new GethLikeBlockNativeTracer(options),
+            { Tracer: var t } when GethLikeNativeTracerFactory.IsNativeTracer(t) => new GethLikeBlockNativeTracer(_worldState, options, new NativeTracerContext(block.Beneficiary)),
```

Comment:
maybe instead of passing `_worldState` and `NativeTracerContext` we should pass a delegate which can produce transaction tracers? (a factory)

Reply:
Would you mind elaborating on this suggestion? A delegate is already being passed in for the transaction tracers in [GethLikeNativeTracerFactory](https://github.com/NethermindEth/nethermind/pull/6907/files/753910358fe324eedc89a8d7195718237f816ab8#diff-0b751946ff1b1722e9ab0a39706ecbcd5e256c47e70d084175ccdf3276f0c1e9R34), are you suggesting that a similar format for registering and building block tracers should be implemented so that world state and context doesn't need to be passed through at the block level?

Reply:
So we can have one Block Tracer class chat would internally use factory to create transaction tracers. Block tracer doesn't have to know about world state at all, only some tx tracers do.